rfc#00003

```diff
@@ -0,0 +1,213 @@
+- tenant
+- host owner (the host operator may be separate, but is considered to
+be the same for this description)
+- entities running processes on the host (includes other workloads
+and compromised processes)
+- man-in-the-middle entities
+Clearly, the first two are the key actors - the others are included for
+reference (no threat model is not defined here).
+
+The following components are defined:
+- host 
+- host CPU + CPU firmware
+- Orchestrator (e.g. Kubernetes, Openshift, Openstack)
+- Tenant workload image
+- Tenant workload image repository
+- Enarx client agent (ECA)
+- Enarx host agent (EHA)
+- Enarx runtime image
+- TEE instance
+- Empty Keep
+- Provisioned Keep
+- Tenant workload
+- Other workloads
+Not all of these exist at all points in the process.
+
+
+### Non-Enarx-enabled use case states
+In the non-TEE-assisted case, all processes running on the host are in
+the host's trust domain (this assumes no compromised processes).  We
+ignore man-in-the-middle entities.  For the default trust domain model,
+we assume that the Orchestrator is operated by the tenant and that the
+Tenant workload image repository is also operated by the tenant.
+
+#### State 0 non-Enarx-enabled use case (initial)
+No Enarx-related components are relevant to this use case.
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Tenant workload image
+  - Tenant workload image repository
+
+Note - there are models where a trust relationship is maintained by
+the tenant to the Tenant workload image repository or to the tenant
+workload, but it is not directly within the trust domain.  These
+models are not examined in this document.
+
+#### State 1 non-Enarx-enabled use case (provisioned)
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Tenant workload
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Tenant workload image
+  - Tenant workload image repository
+The change from state 0 to 1 is the provisioning of the Tenant workload
+onto the host from the Tenant workload image.
+
+The key point of this use case from a trust domain point of view it that
+since the host has access to all processes and associated memory that are
+running on it, all workloads that run on the host can only exist within
+the host trust domain.
+
+### Enarx use case states
+Note that exact state details may vary by implementation.
+
+#### State 0 Enarx use case (initial state)
+Initial state (we assume no compromised processes on host):
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Enarx host agent
+  - Enarx runtime image
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Enarx runtime image
+  - Tenant workload image
+  - Tenant workload image repository
+This is the same as the initial state for the non-Enarx-enabled
+use case.
```

Comment:
```suggestion
use case, excepting the addition of the Enarx host and client agents in their respective trust domains, and the Enarx runtime image, which is in both trust domains.
```

Reply:
Actually, is the Enarx runtime image supposed to be in both?

Reply:
No, it's not: just the host trust domain.  I've adapted the wording slightly, as well.